src/main/java/com/powsybl/caseserver/ObjectStorageService.java

```diff
@@ -0,0 +1,470 @@
+        for (S3Object objectSummary : objectSummaries) {
+            UUID uuid = parseUuidFromKey(objectSummary.key());
+            HeadObjectResponse headObjectResponse = s3Client.headObject(builder -> builder.bucket(bucketName).key(objectSummary.key()));
+            String format = headObjectResponse.metadata().get(FORMAT_HEADER_KEY);  // get format from metadata
+            CaseInfos caseInfos = createInfos(parseFilenameFromKey(objectSummary.key()), uuid, format);
+            caseInfosList.add(caseInfos);
+        }
+        return caseInfosList;
+    }
+
+    @Override
+    public CaseInfos getCase(UUID caseUuid) {
+        var caseFileSummaries = getCaseFileSummaries(caseUuid);
+        if (caseFileSummaries.isEmpty()) {
+            return null;
+        } else {
+            return infosFromDownloadCaseFileSummary(caseFileSummaries.get(0));
+        }
+    }
+
+    @Override
+    public String getCaseName(UUID caseUuid) {
+        CaseInfos caseInfos = getCase(caseUuid);
+        return caseInfos.getName();
+    }
+
+    @Override
+    public Optional<byte[]> getCaseBytes(UUID caseUuid) {
+        String caseFileKey = getCaseFileObjectKey(caseUuid);
+
+        try {
+            GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                    .bucket(bucketName)
+                    .key(caseFileKey)
+                    .build();
+
+            ResponseBytes<GetObjectResponse> objectBytes = s3Client.getObjectAsBytes(getObjectRequest);
+            byte[] content = objectBytes.asByteArray();
+            return Optional.of(content);
+        } catch (NoSuchKeyException e) {
+            LOGGER.error("The expected key does not exist in the bucket s3 : {}", caseFileKey);
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public List<CaseInfos> getCases() {
+        List<S3Object> s3ObjectSummaries = getCasesSummaries(CASES_PREFIX);
+        return infosFromDownloadCaseFileSummaries(s3ObjectSummaries);
+    }
+
+    @Override
+    public boolean caseExists(UUID caseName) {
+        return !getCasesSummaries(uuidToPrefixKey(caseName)).isEmpty();
+    }
+
+    @Override
+    public UUID importCase(MultipartFile mpf, boolean withExpiration) {
+        UUID caseUuid = UUID.randomUUID();
+
+        String caseName = mpf.getOriginalFilename();
+        validateCaseName(caseName);
+
+        String format = withTempCopy(caseUuid, caseName, mpf::transferTo, this::getFormat);
+
+        try (InputStream inputStream = mpf.getInputStream()) {
+            String key = uuidAndFilenameToKey(caseUuid, caseName);
+
+            Map<String, String> userMetadata = new HashMap<>();
+            userMetadata.put(FORMAT_HEADER_KEY, format);
+
+            PutObjectRequest putObjectRequest = PutObjectRequest.builder()
+                    .bucket(bucketName)
+                    .key(key)
+                    .contentType(mpf.getContentType())
+                    .metadata(userMetadata)
```

Comment:
not sure if we want to store the metadata in s3 or in elasticsearch or in postgres hmm

Reply:
We have opted for using Elasticsearch for metadata
done